docs/developer/setup.rst

```diff
@@ -1,100 +1,100 @@
+
+This page describes how to deploy the Renku platform on
+`minikube <https://github.com/kubernetes/minikube>`_.
+Running GitLab (which is a part of Renku) is quite resource
+intensive. For this reason we describe a setup which excludes
+GitLab from the Renku deployment and instead uses `<gitlab.com>`_
+as GitLab backend. For the Renku graph features to function
+properly in this setup, `<gitlab.com>`_ needs to be able to contact
+the locally running platform through a webhook. This can be
+achieved through many services like `ngrok <https://ngrok.com/>`_
+or the like. We are going to use `localhost.run <http://localhost.run>`_,
+a minimalistic method based on ssh tunneling which allows for a somewhat
+static hostname.
 
 The renku source code is hosted on github:
 https://github.com/SwissDataScienceCenter/renku.
 
+
 Prerequisites
--------------
-If you want to start developing the Renku platform, you will need the following
-dependencies:
+^^^^^^^^^^^^^
+You will need the following tools installed on your machine:
 
 .. toctree::
    :maxdepth: 1
 
+   git <https://git-scm.com/>
    python 3.6+ <python_setup>
    pipenv <https://github.com/pypa/pipenv>
+   docker <https://www.docker.com/>
    minikube <https://github.com/kubernetes/minikube>
    kubectl <https://kubernetes.io/docs/tasks/tools/install-kubectl/>
-   helm (>= 2.9.1) <https://github.com/kubernetes/helm/blob/master/docs/install.md>
-   GNU make <https://www.gnu.org/software/make/>
+   helm (^2.9.1) <https://github.com/kubernetes/helm/blob/master/docs/install.md>
+   jq <https://stedolan.github.io/jq/>
 
 For OS X users, we also recommend to install the
 `hyperkit vm driver for minikube <https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperkit-driver>`_.
 
-Getting the code
-----------------
+
+Clone the repo
+^^^^^^^^^^^^^^
+Get the code by cloning the Renku repository:
 
 .. code-block:: console
 
-    $ mkdir src
-    $ cd src
     $ git clone https://github.com/SwissDataScienceCenter/renku.git
-    $ cd renku
-    $ cd charts
+    $ cd renku/charts
 
-``renku`` is the "parent" repository of the collection of microservices, each
-of which has its own repository.
 
-Building and deploying to minikube
-----------------------------------
-
-Install Python dependencies
+Install python dependencies
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-Use ``pipenv`` to manage virtual environment:
+We recommend to use ``pipenv`` for managing your virtual python environment. Install
+all the necessary python dependencies:
 
 .. code-block:: console
 
-    $ pipenv shell
-    $ pipenv install
+    $ pipenv install --dev
 
-Starting minikube
-^^^^^^^^^^^^^^^^^
 
-Using the default VM driver:
+Start minikube
+^^^^^^^^^^^^^^
+Next we start minikube. We recommend using the hyperkit vm driver.
 
 .. code-block:: console
 
-    $ minikube start --memory 6144
+    $ minikube start --memory 6144 --vm-driver hyperkit --kubernetes-version=1.14.8
```

Comment:
I'm not sure we should add the `hyperkit` part, we could have different vm-drivers configured

Reply:
I think it's fine to recommend hyperkit as a default, in my experience it was the least problematic that I've tried.

Reply:
it's specific to macos though isn't it?

Reply:
Oh, is it? Sorry, I missed that part. In this case I agree with both of you that our instructions should use the default VM driver of the users system.

Reply:
I believe in the instructions we used to say that on the mac we've found the hyperkit driver to work best - but I'm not sure if that is actually still the case.

Reply:
In my experience that is still the case. I adapted the wording accordingly (hyperkit recommended for OS X users).